### PR TITLE
Links: table refactor as class component 

### PR DESCRIPTION
**Pull request by Pineirin**
Reformat details links fetch
Add translations to javascript
closes: https://github.com/inveniosoftware/invenio-rdm-records/issues/1087

### CODE REVIEW

**Comment by kpsherva** [comment]
why was the old PR closed?

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/details/LinksTable.js [hunk 1]
```diff
@@ -79,107 +107,111 @@ function LinksTable({ data }) {
...
+    const listRecords = links["oai-listrecords"];
+    const listIdentifiers = links["oai-listidentifiers"];
 
-  // Set default prefix based on backend link, use dublin core if unknown.
-  let defaultPrefix = getPrefixFromLink(listRecords);
-  if (!Object.prototype.hasOwnProperty.call(formats, defaultPrefix)) {
-    defaultPrefix = "oai_dc";
+    return (
+      <>
+        {
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/details/LinksTable.js:145]: redundant { }

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/details/LinksTable.js [hunk 2]
```diff
@@ -79,107 +107,111 @@ function LinksTable({ data }) {
...
-              </Table.Row>
-            </Table.Body>
-          </Table>
-        </>
-      }
-    </>
-  );
 }
 
-LinksTable.propTypes = {
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/details/LinksTable.js:181]: why are props removed? eslint should complain about it, we should add eslint-config-invenio if it is not already there

**Comment by zzacharo** [comment]
@Pineirin we should be careful when we close a [PR](https://github.com/inveniosoftware/invenio-rdm-records/pull/1102) as it is wiping out other people contributions! Make sure in this PR to have the proper co-authors! Out of curiosity, why was it needed to close this?

**Comment by Pineirin** [comment]
> @Pineirin we should be careful when we close a [PR](https://github.com/inveniosoftware/invenio-rdm-records/pull/1102) as it is wiping out other people contributions! Make sure in this PR to set the proper author e.g `git commit --author="John Doe <john@doe.org>"` and add yourself as co-author! Out of curiosity, why was it needed to close this?

I'm not sure if it was needed but I struggled a lot with the permissions for committing the changes so I thought that for simplicity it was best for me to create a clean PR, as I was going to squash the commits anyway.

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/details/LinksTable.js [hunk 3]
```diff
@@ -19,38 +23,65 @@ const knownFormats = {
...
+        this.setState({
+          formats: serialized,
+        });
       }
+    } catch (error) {
+      console.error(error);
+
+      this.setState({
+        error: {
+          header: "Fetch error",
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/details/LinksTable.js:67]: missing translations

**Review by kpsherva — APPROVED** [review]

**Pull request merged**